File: 2. Structural Design Patterns/Composite/Java/CompositeTemplate.java
// Composite Pattern – Java Template

import java.util.*;

abstract class Component { abstract void operation(); }
class Leaf extends Component { @Override void operation() { /* ... */ } }
class Composite extends Component {
    private final List<Component> children = new ArrayList<>();
    void add(Component c) { children.add(c); }
    void remove(Component c) { children.remove(c); }
    @Override void operation() { for (Component c : children) c.operation(); }
}
public class CompositeTemplate {
    public static void main(String[] args){ Composite root = new Composite(); root.add(new Leaf()); root.operation(); }
}

File: 2. Structural Design Patterns/Composite/C#/CompositeTemplate.cs
// Composite Pattern – C# Template
// Component, Leaf, and Composite classes for tree structures.

using System.Collections.Generic;

namespace CompositeTemplate
{
    abstract class Component { public abstract void Operation(); }
    class Leaf : Component { public override void Operation() { /* ... */ } }
    class Composite : Component
    {
        private readonly List<Component> _children = new();
        public void Add(Component c) => _children.Add(c);
        public void Remove(Component c) => _children.Remove(c);
        public override void Operation() { foreach (var child in _children) child.Operation(); }
    }
    class Program { static void Main(){ var comp = new Composite(); comp.Add(new Leaf()); comp.Operation(); } }
}
